src/session/chat/history.rs

```diff
@@ -12,13 +12,14 @@ use crate::utils::do_async;
 
 mod imp {
     use super::*;
+    use glib::WeakRef;
     use once_cell::sync::{Lazy, OnceCell};
     use std::cell::{Cell, RefCell};
     use std::collections::{HashMap, VecDeque};
 
     #[derive(Debug, Default)]
     pub struct History {
-        pub chat: OnceCell<Chat>,
+        pub chat: OnceCell<WeakRef<Chat>>,
```

Comment:
Just a question to see what's your opinion about that:

Now, that we use `WeakRef`, we could remove the `OnceCell` completely and just leave a `WeakRef<Chat>` here to make the code cleaner.
I assume `OnceCell` instead of `RefCell` was used before because the field should only be initialized once. For that, I think the `CONSTRUCT_ONLY` flag is enough for that

This would apply to all other `OnceCell<WeakRef<_>>>`.

Reply:
Yeah, you're right, there's no point on using OnceCell now.